xdsl/irdl/constraints.py

```diff
@@ -339,21 +351,19 @@ def verify(
             exc_msg += "\n".join([str(e) for e in exc_bucket])
             raise VerifyException(exc_msg)
 
-    def get_resolved_variables(self) -> set[str]:
-        if len(self.attr_constrs) == 0:
-            return set()
-        return set[str].union(
-            *[constr.get_resolved_variables() for constr in self.attr_constrs]
-        )
+    def get_resolvers(self) -> dict[str, Resolver[AttributeCovT]]:
+        d: dict[str, Resolver[AttributeCovT]] = dict()
+        for constr in self.attr_constrs:
+            d |= constr.get_resolvers()
```

Comment:
what happens if there are overlapping keys here?

Reply:
Overlapping keys get overwritten

Reply:
Is the idea here also that the expectation is that the resolvers will return identical results, so this is not an issue?

Reply:
Yes